chain/default_syncer.go

```diff
@@ -133,7 +163,11 @@ func (syncer *DefaultSyncer) collectChain(ctx context.Context, tipsetCids types.
 	fetchedHead := tipsetCids
 	defer logSyncer.Infof("chain fetch from network complete %v", fetchedHead)
 
-	for {
+	// Continue collecting the chain if we're either not yet caught up or the
+	// height of the input blocks has not yet exceeded the sum of the current
```

Comment:
Blocking: This comment gives a misleading picture to the reader about what is going on here.  It references the case when the new "too long" chain is built on top of the head of the heaviest chain.  In general the finality check applies to new chains that fork from far back in the past too.  This comment is also confusing because it references the height of the input blocks when the finality limit is based on number of blocks in the code.